crates/gitql-ast/src/function.rs

```diff
@@ -628,6 +629,12 @@ fn date_make_date(inputs: Vec<Value>) -> Value {
     Value::Date(time_stamp)
 }
 
+fn date_dayname(inputs: Vec<Value>) -> Value {
```

Comment:
You need to accept Date not Text so we can make sure it's a valid format, later we support Text to Date Implicit Casting,


The parser error is because you need to register the new function in PROTOTYPE map with the right return type and parameters

https://github.com/AmrDeveloper/GQL/blob/e5d672c3af92e1ed4f0f26cb93d7e379beaef0de/crates/gitql-ast/src/function.rs#L72

Reply:
So the input needs to be a Date value and not a String ?